Example/Tests/GINIMultipageReviewControllerTests.swift

```diff
@@ -0,0 +1,187 @@
+        XCTAssertEqual(thirdCell?.documentImage.image, imageDocuments[2].previewImage,
+                       "Third cell image should match the one passed in the initializer")
+        XCTAssertEqual(firstCell?.documentImage.contentMode, UIViewContentMode.scaleAspectFit,
+                       "First cell content mode should match the one passed in the initializer")
+    }
+    
+    func testMainCollectionCellSize() {
+        let vc = MultipageReviewController(imageDocuments: imageDocuments)
+        _ = vc.view
+        vc.view.setNeedsLayout()
+        vc.view.layoutIfNeeded()
+        
+        let firstCellIndexPath = IndexPath(row: 0, section: 0)
+        let cellSize = vc.collectionView(vc.mainCollection,
+                                         layout: vc.mainCollection.collectionViewLayout,
+                                         sizeForItemAt: firstCellIndexPath)
+        
+        XCTAssertEqual(cellSize, vc.mainCollection.frame.size,
+                       "First cell image should match the one passed in the initializer")
+    }
+    
+    func testMainCollectionInsets() {
+        let collectionInsets = vc.collectionView(vc.mainCollection,
+                                                 layout: vc.mainCollection.collectionViewLayout,
+                                                 insetForSectionAt: 0)
+        
+        XCTAssertEqual(collectionInsets, .zero,
+                       "Main collection insets should be zero")
+    }
+    
+    func testBottomCollectionCellContent() {
+        let firstCell = vc.collectionView(vc.bottomCollection,
+                                          cellForItemAt: IndexPath(row: 0, section: 0)) as? MultipageReviewBottomCollectionCell
+        let secondCell = vc.collectionView(vc.bottomCollection,
+                                           cellForItemAt: IndexPath(row: 1, section: 0)) as? MultipageReviewBottomCollectionCell
+        let thirdCell = vc.collectionView(vc.bottomCollection,
+                                          cellForItemAt: IndexPath(row: 2, section: 0)) as? MultipageReviewBottomCollectionCell
+        XCTAssertEqual(firstCell?.documentImage.image, imageDocuments[0].previewImage,
+                       "First cell image should match the one passed in the initializer")
+        XCTAssertEqual(firstCell?.pageIndicator.text, "1",
+                       "First cell indicator should match its position")
+        XCTAssertEqual(secondCell?.documentImage.image, imageDocuments[1].previewImage,
+                       "Second cell image should match the one passed in the initializer")
+        XCTAssertEqual(secondCell?.pageIndicator.text, "2",
+                       "Second cell indicator should match its position")
+        XCTAssertEqual(thirdCell?.documentImage.image, imageDocuments[2].previewImage,
+                       "Third cell image should match the one passed in the initializer")
+        XCTAssertEqual(thirdCell?.pageIndicator.text, "3",
```

Comment:
Here, and also above, you have the code duplicated three times. First of all, I'd only check two of them. I don't feel that the third check brings any value.

Second of all, if you're going to check all cells all the time, you should find a way to not just copy the code.

Reply:
I check all cells because I'm always a little bit paranoid with "if someone decides to do a check after the second cell that changes something in the next one...". What do you think?